fsharp-kestrel-sample/src/Program.fs

```diff
@@ -1,55 +1,45 @@
-    open Microsoft.Extensions.Logging
-    open Microsoft.Extensions.Configuration
-    open Microsoft.AspNetCore.Builder
-
-    type Startup (env: IHostingEnvironment) =
-        do
-            Log.Logger <- LoggerConfiguration()
-                            .MinimumLevel.Verbose()
-                            .WriteTo.LiterateConsole()
-                            .Enrich.FromLogContext() 
-                            .CreateLogger();
-
-        member this.Configuration
-            with get() = Microsoft.Extensions.Configuration.Memory.MemoryConfigurationSource(
-                            InitialData = (dict [ "Logging:IncludeScopes", "False"
-                                                  "Logging:LogLevel:System", "Warning"
-                                                  "Logging:LogLevel:Default", "Debug"
-                                                  "Logging:LogLevel:Microsoft", "Warning" ]))
-
-        member this.Configure (app : IApplicationBuilder, env: IHostingEnvironment, loggerFactory: ILoggerFactory) =
-            loggerFactory.AddSerilog() |> ignore
-
-            app.Run (fun ctx ->
-                let log = ctx.RequestServices.GetService(typeof<ILogger<unit>>) :?> ILogger<unit>
-                log.LogDebug("got request at {@requestUrl}", ctx.Request.GetEncodedUrl())
-                
-                sprintf "Hello from Serilog ASP.NET Core F# Sample!"
-                |> ctx.Response.WriteAsync
-            )
- 
+namespace FSharpKestrelSample
+
+open System.IO
+open Microsoft.AspNetCore
+open Microsoft.AspNetCore.Hosting
+open Microsoft.Extensions.Configuration
+open Serilog
+
+module Program =
+    let successExitCode = 0
+    let failureExitCode = 1
+
+    let Configuration =
+        ConfigurationBuilder()
+            .SetBasePath(Directory.GetCurrentDirectory())
+            .AddJsonFile("appsettings.json", optional=true, reloadOnChange=true)
+            .Build() :> IConfiguration
+
+    let CreateWebHostBuilder args =
+        WebHost
+            .CreateDefaultBuilder(args)
+            .UseSerilog()
+            .UseStartup<Startup>();
+
     [<EntryPoint>]
-    let main argv = 
-        WebHostBuilder()
-            .UseKestrel()
-            .UseContentRoot(Directory.GetCurrentDirectory())
-            .UseUrls("http://0.0.0.0:5001")
-            .UseStartup<Startup>()
-            .Build()
-            .Run()
-        
-        0 // return an integer exit code
\ No newline at end of file
+    let main args =
+
+        Log.Logger <- LoggerConfiguration()
+            .ReadFrom.Configuration(Configuration)
+            .Enrich.WithProperty("App Name", "Serilog F# Kestrel Sample")
+            .CreateLogger();
+
+        Log.Information("Starting with arguments {Args}", args)
+
+        try
+            try
+                CreateWebHostBuilder(args).Build().Run()
+                successExitCode
+            with
+            | ex ->
+                Log.Fatal(ex, "Host terminated unexpectedly");
```

Comment:
your C# is showing ;)

Reply:
Yes, I was trying to keep it similar, sorry ;)

Reply:
`;`

Reply:
Ahh! Good catch, thanks 👍